include/api/CForecastRunner.h

```diff
@@ -68,10 +69,22 @@ class API_EXPORT CForecastRunner final : private core::CNonCopyable {
     static const size_t DEFAULT_EXPIRY_TIME = 14 * core::constants::DAY;
 
     //! max memory allowed to use for forecast models
-    static const size_t MAX_FORECAST_MODEL_MEMORY = 20971520; // 20MB
+    static const size_t MAX_FORECAST_MODEL_MEMORY = 20971520ull; // 20MB
+
+    // Note: This value measures the size in memory, not the size of the persistence,
+    // which is likely higher and would be hard to calculate upfront
+    //! max memory allowed to use for forecast models persisting to disk
+    static const size_t MAX_FORECAST_MODEL_PERSISTANCE_MEMORY = 524288000ull; // 500MB
+
+    // Note: This value is lower than on X-pack side to prevent side-effects,
+    // if you change this value also change the limit on X-pack side.
+    // The purpose of this value is to guard the rest of the system regarding
+    // an out of disk space
+    //! minimum disk space required for disk persistence
```

Comment:
Only the last line of this begins with `//!`.  All the lines should be consistent.